Index: src/org/eclipse/epsilon/evl/emf/validation/EmfMarkerResolver.java
===================================================================
--- src/org/eclipse/epsilon/evl/emf/validation/EmfMarkerResolver.java	(revision 524)
+++ src/org/eclipse/epsilon/evl/emf/validation/EmfMarkerResolver.java	(working copy)
@@ -10,10 +10,13 @@
  ******************************************************************************/
 package org.eclipse.epsilon.evl.emf.validation;
 
+import java.net.URL;
+
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IMarker;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.FileLocator;
 import org.eclipse.core.runtime.Path;
 import org.eclipse.emf.common.util.URI;
 import org.eclipse.emf.ecore.EObject;
@@ -50,9 +55,15 @@
 	}
 	
 	public EditingDomain getEditingDomain(IMarker marker) {
-		String filePath = getElementResourceLocation(marker);
-		IFile file = ResourcesPlugin.getWorkspace().getRoot().getFile(new Path(getElementResourceLocation(marker)));
-		String editorId = PlatformUI.getWorkbench().getEditorRegistry().getDefaultEditor(filePath).getId();
+		String sResourceLocation = getElementResourceLocation(marker);
+		String sFilePath;
+		try {
+			sFilePath = FileLocator.toFileURL(new URL(sResourceLocation)).getPath();
+		} catch (Exception e) {
+			sFilePath = sResourceLocation;
+		}
+		IFile file = ResourcesPlugin.getWorkspace().getRoot().getFileForLocation(new Path(sFilePath));
+		String editorId = PlatformUI.getWorkbench().getEditorRegistry().getDefaultEditor(sResourceLocation).getId();
 		
 		//PlatformUI.getWorkbench().getEditorRegistry().
 		
Index: src/org/eclipse/epsilon/evl/emf/validation/GmfMarkerResolver.java
===================================================================
--- src/org/eclipse/epsilon/evl/emf/validation/GmfMarkerResolver.java	(revision 524)
+++ src/org/eclipse/epsilon/evl/emf/validation/GmfMarkerResolver.java	(working copy)
@@ -73,7 +76,7 @@
 		
 			
 		//URI uri = URI.createURI(marker.getResource().getFullPath().toString());
-		String location = "platform:" + marker.getResource().getFullPath().toString();
+		String location = "platform:/resource" + marker.getResource().getFullPath().toString();
 		
 		//System.err.println("GMF : " + location);
 		
